py-rse/git-advanced.Rmd

```diff
@@ -245,13 +246,16 @@ In order to determine the best method for estimating \(\alpha\) and fitting a li
 we turn to a scientific paper on Zipf’s Law in english texts [@Moreno2016].
 The mathematics in the paper is a little heavy,
 but the authors essentially suggest using a method called maximum likelihood estimation. 
-The likelihood is the probability of our observed data, 
+The likelihood function is the probability of our observed data, 
 viewed as a function of the parameters in the statistical model that we assume generated it. 
 In maximum likelihood estimation, we estimate the parameters in the model 
 by choosing them to maximize this likelihood, although
 computationally, it is often easier to minimize the negative log likelihood function.
 @Moreno2016 define the likelihood using a parameter, \(\beta\), 
-that we will see below is directly related to the \(\alpha\) parameter in our definition of Zipf's Law.
+which is directly related to the \(\alpha\) parameter in our definition of Zipf's Law,
+through \(\alpha = \tfrac{1}{\beta-1}\).
+Under their model, the value of $c$ is the total number of unique words, or 
+equivalently the largest value of the rank.
```

Comment:
For this hypothetical word count distribution,
```
the: 5 times (rank 1)
at: 4 times (rank 2)
hello: 1 time (rank 3)
goodbye: 1 time (rank 3)
```
the largest value of the rank is 3 but the total number of unique words is 4. This isn't a big issue for large texts (the difference between the max rank and number of unique words will be relatively small) but to be correct should be use the number of unique words and not the maximum rank in our code?

Reply:
Depends on your definition of rank.  In R, your definition is the min rank:
``` r
rank(dplyr::desc(c(5, 4, 1, 1)), ties.method = "min")
#> [1] 1 2 3 3
```

We should be using the max rank:
```r
rank(dplyr::desc(c(5, 4, 1, 1)), ties.method = "max")
#> [1] 1 2 4 4
```
With max rank, the maximum max rank is always the same as number of values to rank.

I'm not sure what is the default in python, but we need to check.

Reply:
By default pandas was using the average rank, so I've changed it to max and updated the various scripts and chapters accordingly